tutorials/tts/FastPitch_Adapter_Finetuning.ipynb

```diff
@@ -0,0 +1,877 @@
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "47c5983a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wave_model = WaveformFeaturizer(sample_rate=sample_rate)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d0a71553",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# FastPitch\n",
+    "spec_model = FastPitchModel.load_from_checkpoint(pretrained_fastpitch_checkpoint)\n",
+    "spec_model.load_adapters(finetuned_adapter_checkpoint)\n",
+    "# spec_model.freeze()\n",
+    "# spec_model.unfreeze_enabled_adapters()\n",
+    "spec_model = spec_model.eval().cuda()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8ccaf4e2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# HiFiGAN\n",
+    "vocoder_model = HifiGanModel.load_from_checkpoint(checkpoint_path=finetuned_hifigan_on_adaptation_checkpoint).eval().cuda()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "eebc7522",
+   "metadata": {},
+   "source": [
+    "## b. Output Audio"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "035cacf7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def gt_spectrogram(audio_path, wave_model, spec_gen_model):\n",
+    "    features = wave_model.process(audio_path, trim=False)\n",
+    "    audio, audio_length = features, torch.tensor(features.shape[0]).long()\n",
+    "    audio = audio.unsqueeze(0).to(device=spec_gen_model.device)\n",
+    "    audio_length = audio_length.unsqueeze(0).to(device=spec_gen_model.device)\n",
+    "    with torch.no_grad():\n",
+    "        spectrogram, spec_len = spec_gen_model.preprocessor(input_signal=audio, length=audio_length)\n",
+    "    return spectrogram, spec_len\n",
+    "\n",
+    "def gen_spectrogram(text, spec_gen_model, reference_spec, reference_spec_lens):\n",
+    "    parsed = spec_gen_model.parse(text)\n",
+    "    with torch.no_grad():    \n",
+    "        spectrogram = spec_gen_model.generate_spectrogram(tokens=parsed, \n",
+    "                                                          reference_spec=reference_spec, \n",
+    "                                                          reference_spec_lens=reference_spec_lens)\n",
+    "\n",
+    "    return spectrogram\n",
+    "  \n",
+    "def synth_audio(vocoder_model, spectrogram):    \n",
+    "    with torch.no_grad():  \n",
+    "        audio = vocoder_model.convert_spectrogram_to_audio(spec=spectrogram)\n",
+    "    if isinstance(audio, torch.Tensor):\n",
+    "        audio = audio.to('cpu').numpy()\n",
+    "    return audio"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5a187093",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Reference Audio\n",
+    "with open(train_manifest, \"r\") as f:\n",
+    "    for i, line in enumerate(f):\n",
+    "        reference_record = json.loads(line)\n",
+    "        break\n",
```

Comment:
Is this randomly selecting the first line from the training data as the reference? Shouldn't the reference come from the validation set, and belong to the same speaker?

Reply:
I expect user only use one target speaker data to run this notebook, so the training set and validation set are from the same speaker. I think there is no much difference we use from training or validation set.

Reply:
Got it, that makes sense for adapters.